docs/csharp/programming-guide/concepts/async/how-to-extend-the-async-walkthrough-by-using-task-whenall.md

```diff
@@ -1,438 +1,338 @@
+# Extend the async tutorial using Task.WhenAll (C#)
 
-You might have noticed in the walkthrough that the websites download at different rates. Sometimes one of the websites is very slow, which delays all the remaining downloads. When you run the asynchronous solutions that you build in the walkthrough, you can end the program easily if you don't want to wait, but a better option would be to start all the downloads at the same time and let faster downloads continue without waiting for the one that’s delayed.
+You can improve the performance of the async solution in [Tutorial: Access the web with async and await (C#)](walkthrough-accessing-the-web-by-using-async-and-await.md) by using the <xref:System.Threading.Tasks.Task.WhenAll%2A?displayProperty=nameWithType> method. This method asynchronously awaits multiple asynchronous operations, which are represented as a collection of tasks.
 
-You apply the `Task.WhenAll` method to a collection of tasks. The application of `WhenAll` returns a single task that isn’t complete until every task in the collection is completed. The tasks appear to run in parallel, but no additional threads are created. The tasks can complete in any order.
+You might have noticed in the tutorial that the websites download at different rates. Sometimes one of the websites is slow, which delays all the remaining downloads. When you run the asynchronous solutions that you build in the walkthrough, you can end the program easily if you don't want to wait, but a better option would be to start all the downloads at the same time and let faster downloads continue without waiting for the one that's delayed.
+
+You apply the `Task.WhenAll` method to a collection of tasks. The application of `WhenAll` returns a single task that isn't complete until every task in the collection is completed. The tasks appear to run in parallel, but no additional threads are created. The tasks can complete in any order.
 
 > [!IMPORTANT]
-> The following procedures describe extensions to the async applications that are developed in [Walkthrough: Accessing the Web by Using async and await (C#)](./walkthrough-accessing-the-web-by-using-async-and-await.md). You can develop the applications by either completing the walkthrough or downloading the code from [Developer Code Samples](https://code.msdn.microsoft.com/Async-Sample-Accessing-the-9c10497f).
+> The following procedures describe extensions to the async applications that are developed in [Tutorial: Access the web with async and await (C#)](walkthrough-accessing-the-web-by-using-async-and-await.md). You can develop the applications by either completing the tutorial or downloading the code from [Code samples browser](https://docs.microsoft.com/samples/dotnet/samples/async-and-await-cs).
```

Comment:
I think the old link refers to https://github.com/dotnet/docs/issues/13682, while the new link refers to https://github.com/dotnet/docs/issues/13680, which are different code samples.

Reply:
The old link was referring to a single method in an old sample source code project. I'm updating this to based on extended a migrated tutorial, and extending that specific tutorial.